### PR TITLE
Add removeHeaders to virtualservice

### PR DESCRIPTION
**Pull request by cmluciano**
Related https://github.com/istio/istio/issues/5639

### CODE REVIEW

**Comment by cmluciano** [comment]
cc @GregHanson @ijsnellf

**Comment by cmluciano** [comment]
@rshriram Clarified in the comments. Should the actual field be called remove_response_headers as well?

File: networking/v1alpha3/virtual_service.proto [hunk 1]
```diff
@@ -380,6 +380,10 @@ message HTTPRoute {
   // Additional HTTP headers to add before forwarding a request to the
   // destination service.
   map<string, string> append_headers = 11;
+  // Additional HTTP headers to remove before forwarding a response to the
+  // destination service.
+  // $hide_from_docs
```
Comment (rshriram) [inline networking/v1alpha3/virtual_service.proto]: Remove response headers 
Http headers to remove before returning the response to the caller

File: networking/v1alpha3/virtual_service.pb.go [hunk 2]
```diff
@@ -387,6 +387,9 @@ type HTTPRoute struct {
 	// Additional HTTP headers to add before forwarding a request to the
 	// destination service.
 	AppendHeaders map[string]string `protobuf:"bytes,11,rep,name=append_headers,json=appendHeaders" json:"append_headers,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
+	// Http headers to remove before returning the response to the caller
```
Comment (frankbu) [inline networking/v1alpha3/virtual_service.pb.go]: nit: leave a blank line before this one

Comment (cmluciano) [reply networking/v1alpha3/virtual_service.pb.go]: fixed

**Pull request merged**